File: server/models/Skill.js
const mongoose = require('mongoose');

const SkillSchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true },
  language: { type: String, default: 'javascript' }, // 'javascript', 'typescript', 'css', etc.
  code: String, // The code snippet
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Skill', SkillSchema);

File: backend/src/main/java/com/portfolio/entity/Skill.java
package com.portfolio.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Getter @Setter
@Table(name = "skills")
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;

    private String language; // 'javascript', 'typescript', 'css', etc.

    @Column(columnDefinition = "TEXT")
    private String code;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
